src/IO/HDF5/ParallelHDF5IOHandler.cpp

```diff
@@ -60,6 +61,37 @@ ParallelHDF5IOHandlerImpl::ParallelHDF5IOHandlerImpl(
 {
     m_datasetTransferProperty = H5Pcreate(H5P_DATASET_XFER);
     m_fileAccessProperty = H5Pcreate(H5P_FILE_ACCESS);
+    m_fileCreateProperty = H5Pcreate(H5P_FILE_CREATE);
+
+    auto const hdf5_spaced_allocation = auxiliary::getEnvString( "OPENPMD_HDF5_PAGED_ALLOCATION", "ON" );
+    if( hdf5_spaced_allocation == "ON" ) {
+        auto const strPageSize = auxiliary::getEnvString( "OPENPMD_HDF5_PAGED_ALLOCATION_SIZE", "33554432" );
+        std::stringstream tstream(strPageSize);
+        hsize_t page_size;
+        tstream >> page_size;
+
+        H5Pset_file_space_strategy(m_fileCreateProperty, H5F_FSPACE_STRATEGY_PAGE, 0, (hsize_t)0);
```

Comment:
Ups, we need to guard this one as well with a macro for the HDF5 version :) @jeanbez 

Seen on nightly install CI:
```
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp: In constructor ‘openPMD::ParallelHDF5IOHandlerImpl::ParallelHDF5IOHandlerImpl(openPMD::AbstractIOHandler*, MPI_Comm, nlohmann::json)’:
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp:73:58: error: ‘H5F_FSPACE_STRATEGY_PAGE’ was not declared in this scope
         H5Pset_file_space_strategy(m_fileCreateProperty, H5F_FSPACE_STRATEGY_PAGE, 0, (hsize_t)0);
                                                          ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp:73:58: note: suggested alternative: ‘H5D_SPACE_STATUS_ERROR’
         H5Pset_file_space_strategy(m_fileCreateProperty, H5F_FSPACE_STRATEGY_PAGE, 0, (hsize_t)0);
                                                          ^~~~~~~~~~~~~~~~~~~~~~~~
                                                          H5D_SPACE_STATUS_ERROR
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp:73:9: error: ‘H5Pset_file_space_strategy’ was not declared in this scope
         H5Pset_file_space_strategy(m_fileCreateProperty, H5F_FSPACE_STRATEGY_PAGE, 0, (hsize_t)0);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp:73:9: note: suggested alternative: ‘H5Pset_file_space’
         H5Pset_file_space_strategy(m_fileCreateProperty, H5F_FSPACE_STRATEGY_PAGE, 0, (hsize_t)0);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
         H5Pset_file_space
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp:74:9: error: ‘H5Pset_file_space_page_size’ was not declared in this scope
         H5Pset_file_space_page_size(m_fileCreateProperty, page_size);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/ci-x1pUtoapBt/src/src/IO/HDF5/ParallelHDF5IOHandler.cpp:74:9: note: suggested alternative: ‘H5Pset_file_space’
         H5Pset_file_space_page_size(m_fileCreateProperty, page_size);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
         H5Pset_file_space
```
With HDF5 1.10.0.1:
```
-- Found HDF5: /usr/lib/x86_64-linux-gnu/hdf5/openmpi/libhdf5.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so (found version "1.10.0.1") found components: C 
```

Reply:
Looks like we just need to guard this for the HDF5 1.10.1+ release :)
https://support.hdfgroup.org/HDF5/doc/RM/H5P/H5Pset_file_space_strategy.htm